src/shader.cpp

```
@@ -204,6 +208,165 @@ const ShaderProgramDesc& ShaderProgram::getDesc()
+                record.scope = src.scope;
+                record.access = src.access;
+                record.entryPointIndex = src.entryPointIndex;
+                record.space = src.space;
+                record.binding = src.binding;
+                record.uniformOffset = src.uniformOffset;
+                record.uniformStride = src.uniformStride;
+                if (src.debugName)
+                    record.debugName = src.debugName;
+
+                SLANG_RETURN_ON_FAIL(_validateSyntheticResourceRecord(record));
+
```

Cppcheck (2.20.0)

[portability] Shifting signed 32-bit value by 31 bits is implementation-defined behaviour

(shiftTooManyBitsSigned)